src/network/mcpe/handler/DeathPacketHandler.php

```diff
@@ -44,6 +48,16 @@ public function setUp() : void{
 			RespawnPacket::SEARCHING_FOR_SPAWN,
 			$this->player->getId()
 		));
+
+		/** @var string[] $parameters */
+		$parameters = [];
+		if($this->deathMessage instanceof Translatable){
+			$message = $this->deathMessage->getText();
```

Comment:
This isn't reliable. The string might not be present in the vanilla files.

You should implement similar logic to what's done here: https://github.com/pmmp/PocketMine-MP/blob/fe982c697bec3265c4222b73fe5ed17c9ef3b1ce/src/player/Player.php#L1985-L1993